Tests/Unit/Content/ContentTypeResolver/TeaserSelectionResolverTest.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\HeadlessBundle\Tests\Unit\Content\ContentTypeResolver;
+
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use Sulu\Bundle\HeadlessBundle\Content\ContentTypeResolver\TeaserSelectionResolver;
+use Sulu\Bundle\HeadlessBundle\Content\Serializer\TeaserSerializerInterface;
+use Sulu\Bundle\PageBundle\Teaser\Teaser;
+use Sulu\Bundle\PageBundle\Teaser\TeaserManagerInterface;
+use Sulu\Component\Content\Compat\PropertyInterface;
+
+class TeaserSelectionResolverTest extends TestCase
+{
+    /**
+     * @var TeaserManagerInterface|ObjectProphecy
+     */
+    private $teaserManager;
+
+    /**
+     * @var TeaserSerializerInterface|ObjectProphecy
+     */
+    private $teaserSerializer;
+
+    /**
+     * @var TeaserSelectionResolver
+     */
+    private $teaserSelectionResolver;
+
+    protected function setUp(): void
+    {
+        $this->teaserManager = $this->prophesize(TeaserManagerInterface::class);
+        $this->teaserSerializer = $this->prophesize(TeaserSerializerInterface::class);
+
+        $this->teaserSelectionResolver = new TeaserSelectionResolver(
+            $this->teaserManager->reveal(),
+            $this->teaserSerializer->reveal()
+        );
+    }
+
+    public function testGetContentType(): void
+    {
+        self::assertSame('teaser_selection', $this->teaserSelectionResolver::getContentType());
+    }
+
+    public function testResolve(): void
+    {
+        $locale = 'en';
+        $items = [
+            [
+                'id' => '74a36ca1-4805-48a0-b37d-3ffb3a6be9b1',
+                'type' => 'pages',
```

Comment:
its a little bit strange from outside that we return here `pluralized` types and in the structure itself it are the singular template key but will at current state create an issue for it. https://github.com/sulu/SuluHeadlessBundle/issues/84 /cc @nnatter

Reply:
Yes I totally agree